src/sql/workbench/browser/designer/designer.ts

```diff
@@ -836,6 +824,53 @@ export class Designer extends Disposable implements IThemable {
 					headerRowHeight: TableHeaderRowHeight,
 					editorLock: new Slick.EditorLock()
 				});
+
+				let moveRowsPlugin = new RowMoveManager({
+					cancelEditOnDrag: true,
+					id: 'moveRow',
+					iconCssClass: Codicon.grabber.classNames,
+					title: localize('designer.moveRowText', 'Move Row'),
+					width: 20,
+					resizable: false,
+					isFontIcon: true,
+					behavior: 'selectAndMove'
+				});
+				moveRowsPlugin.onBeforeMoveRows.subscribe((e, data) => {
```

Comment:
what are these for? a lot of event subscriptions that do nothing.

Reply:
these are the subscriptions for the drag and drop icon. I'll be updating it with Hai's changes, this isn't complete yet.